File: arena-jface/src/main/java/org/uqbar/lacar/ui/impl/jface/tree/ReflectionLabelProvider.java
package org.uqbar.lacar.ui.impl.jface.tree;

import org.eclipse.jface.viewers.LabelProvider;
import org.uqbar.commons.utils.ReflectionUtils;

public class ReflectionLabelProvider extends LabelProvider {
	
	private final String propertyName;

	public ReflectionLabelProvider(String propertyName) {
		this.propertyName = propertyName;
	}
	
	@Override
	public String getText(Object element) {
		if(propertyName != null){
			return ReflectionUtils.invokeGetter(element, propertyName) + "";
		}else{
			return element.toString();
		}
	}

}

File: arena-jface/src/main/scala/org/uqbar/eclipse/jface/ReflectionLabelProvider.scala
package org.uqbar.eclipse.jface

import org.eclipse.jface.viewers.LabelProvider
import org.uqbar.commons.utils.ReflectionUtils

class ReflectionLabelProvider(var propertyName:String) extends LabelProvider {

  override def getText(element:Object) = 
    if (propertyName != null) 
	  ReflectionUtils.invokeGetter(element, propertyName) + ""
    else
      element.toString
  
}